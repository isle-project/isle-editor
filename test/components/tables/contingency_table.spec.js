// MODULES //

import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import ContingencyTable from '@isle-project/components/tables/contingency-table';
import heartdisease from './../fixtures/heartdisease.json';


// TESTS //

describe( '<ContingencyTable />', function test() {
	it( 'renders a contingency table', () => {
		const table = renderer.create( <ContingencyTable
			data={heartdisease}
			rowVar="Gender"
			colVar="Drugs"
		/> );
		expect( table ).toMatchSnapshot();
	});
});
