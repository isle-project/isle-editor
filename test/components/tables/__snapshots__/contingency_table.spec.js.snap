// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContingencyTable /> renders a contingency table 1`] = `
<table
  className="sortable-theme-bootstrap table table-sm table-bordered"
  data-sortable={true}
>
  <thead>
    <tr>
      <th>
        Gender
         \\ 
        Drugs
      </th>
      <th>
        0
      </th>
      <th>
        1
      </th>
      <th>
        2
      </th>
      <th>
        row-totals
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>
        Female
      </th>
      <td>
        477
      </td>
      <td>
        63
      </td>
      <td>
        68
      </td>
      <td>
        608
      </td>
    </tr>
    <tr>
      <th>
        Male
      </th>
      <td>
        133
      </td>
      <td>
        26
      </td>
      <td>
        21
      </td>
      <td>
        180
      </td>
    </tr>
  </tbody>
  <tbody>
    <tr>
      <th>
        column-totals
      </th>
      <td>
        610
      </td>
      <td>
        89
      </td>
      <td>
        89
      </td>
      <th>
        788
      </th>
    </tr>
  </tbody>
</table>
`;
